### PR TITLE
[non-breaking] deps(udp): make tracing optional and add optional log

### PR DESCRIPTION
**Pull request by mxinden**
Alternative to https://github.com/quinn-rs/quinn/pull/1921.

Implementation of @Ralith suggestion in https://github.com/quinn-rs/quinn/pull/1921#issuecomment-2237230567.

Not breaking `quinn-udp`. I.e. only bumping patch version to `v0.5.3`.

### CODE REVIEW

File: quinn-udp/Cargo.toml [hunk 1]
```diff
@@ -14,14 +14,17 @@ workspace = ".."
 all-features = true
 
 [features]
-default = ["log"]
-# Write logs via the `log` crate when no `tracing` subscriber exists
+default = ["tracing", "log"]
+# Write logs via the `log` crate trough the `tracing` crate when no `tracing`
```
Comment (djc) [inline quinn-udp/Cargo.toml:18]: Maybe shorten this a bit to "Configure tracing to log events via `log` if no `tracing` subscriber exists"?

File: quinn-udp/Cargo.toml [hunk 2]
```diff
@@ -14,14 +14,17 @@ workspace = ".."
...
+# subscriber exists
 log = ["tracing/log"]
+native-log = ["dep:log"]
```
Comment (djc) [inline quinn-udp/Cargo.toml:21]: Bikeshedding a little, I think I like `direct-log` better? Thoughts?

**Review by djc — APPROVED** [review]
LGTM!

File: quinn-udp/Cargo.toml (hunk 1 above)
Comment (mxinden) [reply quinn-udp/Cargo.toml]: Works for me. I updated the comment across all `Cargo.toml` files to stay consistent. Let me know what you think @djc.

File: quinn-udp/Cargo.toml (hunk 2 above)
Comment (mxinden) [reply quinn-udp/Cargo.toml]: Both fine with me. Latest push updates to `direct-log`.

**Review by Ralith — APPROVED** [review]
LGTM, thanks!

**Comment by mxinden** [comment]
Thank you @djc and @Ralith for the **very** quick help and fix here!

This is a great step forward for me. The only audits remaining for `quinn-udp` to land in Firefox is `quinn-udp` itself :tada: 

Could you release `quinn-udp` `v0.5.3` sometime this week?

**Comment by djc** [comment]
Sure, want to submit a PR to bump the version number?

**Comment by mxinden** [comment]
@djc already done in this pull request:

https://github.com/quinn-rs/quinn/blob/main/quinn-udp/Cargo.toml#L3

**Comment by djc** [comment]
Ah sorry, was looking at an outdated checkout. I'll just publish it now then!

**Comment by djc** [comment]
* Published quinn-udp v0.5.3 at registry `crates-io`
* [new tag]           quinn-udp-0.5.3 -> quinn-udp-0.5.3

**Comment by mxinden** [comment]
:pray: thanks again for the quick help @djc!

**Pull request merged**